### PR TITLE
[BUD-44] Login Front end logic

### PR DESCRIPTION
**Pull request by lucianojss**
# Login Front end logic
# [#BUD-44](https://digitalrig.atlassian.net/browse/BUD-XX) 

## Description

Login screen logic, includes form, GraphQl login service call, token storage in cookies, error validations.

## CheckList

- [x] Apollo Client login call
- [x] Form validations client side
- [x] Network and service errors handling
- [x] ErrorDialog component
- [ ] Unit tests 

# Screenshots

Screenshot 2019-10-16 at 14 41 52
Screenshot 2019-10-16 at 14 41 41
Screenshot 2019-10-16 at 14 41 26

### CODE REVIEW

File: src/components/Login/__tests__/Login.test.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  test('should render loading state initially after submitting form', () => {
+    const component = create(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <MemoryRouter initialEntries={[mockLocation]}>
+          <Login />
+        </MemoryRouter>
+      </MockedProvider>
+    );
+
+    // // find the button and simulate a click
```
[the remark this answers is not in the document]
Comment (lucianojss) [reply src/components/Login/__tests__/Login.test.tsx]: This branch stills a work in progress, that's why there is some dirty comments :) 
I will remove them

File: src/styles/theme.ts [hunk 2]
```diff
@@ -4,15 +4,6 @@ import { colorSchemes } from '@buildit/gravity-particles';
 const { tealWhite } = colorSchemes;
 
 let theme = createMuiTheme({
-  breakpoints: {
```
[the remark this answers is not in the document]
Comment (lucianojss) [reply src/styles/theme.ts]: Yes true, I been searching and the default behaviour on major webapps is to not define a minimum value for the screen, so basically I just removed that minimum 320px size. Now it's back to the default so the app can reduce in size to 0px.

If you think it doesn't make sense and it was good as it was before I will just put it back 👍

File: src/utils/auth.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import Cookies from 'js-cookie';
+
+enum AUTH {
+  TOKEN = '_t',
+}
+
+class Auth {
+  getToken() {
```
[the remark this answers is not in the document]
Comment (lucianojss) [reply src/utils/auth.ts]: @Tonik1204 give us a hint here 👍

[the remark this answers is not in the document]
Comment (Tonik1204) [reply src/utils/auth.ts]: Yes, that should be something like this: 

1) For `auth.ts`:
```
const tokenName = process.env.REACT_APP_AUTH_TOKEN || AUTH.TOKEN;
```
2) And for `apollo-client.ts`:
```
import auth from './auth';
const token = auth.getToken();
```

File: src/components/Login/Login.tsx [hunk 4]
```diff
@@ -1,70 +1,153 @@
...
+    margin: '0 auto',
     marginTop: theme.spacing(3),
   },
   submit: {
     margin: theme.spacing(3, 0, 2),
   },
 }));
 
-export default function SignIn() {
+interface FormData {
```
[the remark this answers is not in the document]
Comment (lucianojss) [reply src/components/Login/Login.tsx]: fixed

File: src/components/Login/login.dictionary.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+interface LOGIN_DICTIONARY {
```
[the remark this answers is not in the document]
Comment (lucianojss) [reply src/components/Login/login.dictionary.ts]: fixed

**Pull request merged**